.github/workflows/az-deploy.yml

```diff
@@ -0,0 +1,37 @@
+name: Deploy Azure Resources
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Setup Azure CLI
+        run: |
+          curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+          az --version  # Verify installation
+
+      - name: Login to Azure
+        run: |
+          az login --service-principal -u ${{ secrets.AUTO_AZURE_CLIENT_ID }} -p ${{ secrets.AUTO_AZURE_CLIENT_SECRET }} --tenant ${{ secrets.AUTO_AZURE_TENANT_ID }}
```

Comment:
We've created new variables `AUTO_` - and not re-using the existing ones - we should update the existing github variables to your new subscription values and use them. At the moment, this strikes me as making things more complicated

Reply:
For testing purposes, we have created these variables so that they won't affect the current deployment. In the end, we will be using the same variables.

Reply:
The current deployment is not working as the spn has expired - you are free to re-use those variables if you like